src/components/Monaco.vue

```diff
@@ -3,100 +3,108 @@
 </template>
 
 <script>
-import * as monaco from 'monaco-editor'
-import { computed, onMounted, onUnmounted, ref, watch } from 'vue'
-// import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
-// import { createMonacoComplete, createMonacoHover } from "@/common/monacoUtils";
+import { defineComponent } from 'vue'
 
-export default {
+export default defineComponent({
   name: 'Monaco',
+})
+</script>
 
-  props: {
-    id: {
-      type: String,
-      required: true,
-    },
-    value: {
-      type: String,
-      required: true,
-    },
-    lang: {
-      type: String,
-      required: true,
-    },
-    disabled: {
-      type: Boolean,
-      default: false,
-    },
-    // warp: {
-    //   type: Boolean,
-    //   default: false,
-    // },
-    // provider: {
-    //   type: Array,
-    //   default: () => [],
-    // },
-    scrollLoading: {
-      type: Boolean,
-      default: false,
-    },
-    scrollFunc: {
-      type: Function,
-      default: () => () => {},
-    },
+<script setup>
+import * as monaco from 'monaco-editor'
+import { defineProps, defineEmits, onMounted, onUnmounted, watch } from 'vue'
+
+const prop = defineProps({
+  id: {
+    type: String,
+    required: true,
   },
-  setup(prop) {
-    const editor = ref({})
-    // const editValue = computed(() => {
-    //   console.log(prop.value + "changed");
-    //   return prop.value;
-    // });
+  modelValue: {
+    type: String,
+    required: true,
+  },
+  lang: {
+    type: String,
+    required: true,
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  // warp: {
+  //   type: Boolean,
+  //   default: false,
+  // },
+  // provider: {
+  //   type: Array,
+  //   default: () => [],
+  // },
```

Comment:
Should we need these options? I think we should.

Reply:
I haven't figured out the usage of these two yet, I'll add them later.